Content.Server/Radiation/Systems/GeigerSystem.cs

```diff
@@ -0,0 +1,190 @@
+using Content.Server.Radiation.Components;
+using Content.Server.Radiation.Events;
+using Content.Shared.Hands;
+using Content.Shared.Interaction;
+using Content.Shared.Inventory.Events;
+using Content.Shared.Radiation.Components;
+using Content.Shared.Radiation.Systems;
+using Robust.Server.GameObjects;
+using Robust.Shared.GameStates;
+using Robust.Shared.Player;
+using Robust.Shared.Random;
+
+namespace Content.Server.Radiation.Systems;
+
+public sealed class GeigerSystem : SharedGeigerSystem
+{
+    [Dependency] private readonly SharedAppearanceSystem _appearance = default!;
+    [Dependency] private readonly AudioSystem _audio = default!;
+    [Dependency] private readonly IRobustRandom _random = default!;
+    [Dependency] private readonly RadiationSystem _radiation = default!;
+
+    private static readonly float ApproxEqual = 0.01f;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<GeigerComponent, ActivateInWorldEvent>(OnActivate);
+
+        SubscribeLocalEvent<GeigerComponent, GotEquippedEvent>(OnEquipped);
+        SubscribeLocalEvent<GeigerComponent, GotEquippedHandEvent>(OnEquippedHand);
+        SubscribeLocalEvent<GeigerComponent, GotUnequippedEvent>(OnUnequipped);
+        SubscribeLocalEvent<GeigerComponent, GotUnequippedHandEvent>(OnUnequippedHand);
+
+        SubscribeLocalEvent<RadiationSystemUpdatedEvent>(OnUpdate);
+        SubscribeLocalEvent<GeigerComponent, ComponentGetState>(OnGetState);
+    }
+
+    private void OnActivate(EntityUid uid, GeigerComponent component, ActivateInWorldEvent args)
+    {
+        if (args.Handled || component.AttachedToSuit)
+            return;
+        args.Handled = true;
+
+        SetEnabled(uid, component, !component.IsEnabled);
+    }
+
+    private void OnEquipped(EntityUid uid, GeigerComponent component, GotEquippedEvent args)
+    {
+        if (component.AttachedToSuit)
+            SetEnabled(uid, component, true);
+        SetUser(uid, component, args.Equipee);
+    }
+
+    private void OnEquippedHand(EntityUid uid, GeigerComponent component, GotEquippedHandEvent args)
+    {
+        if (component.AttachedToSuit)
+            return;
+
+        SetUser(uid, component, args.User);
+    }
+
+    private void OnUnequipped(EntityUid uid, GeigerComponent component, GotUnequippedEvent args)
+    {
+        if (component.AttachedToSuit)
+            SetEnabled(uid, component, false);
+        SetUser(uid, component, null);
+    }
+
+    private void OnUnequippedHand(EntityUid uid, GeigerComponent component, GotUnequippedHandEvent args)
+    {
+        if (component.AttachedToSuit)
+            return;
+
+        SetUser(uid, component, null);
+    }
+
+    private void OnUpdate(RadiationSystemUpdatedEvent ev)
+    {
+        var query = EntityQuery<GeigerComponent, RadiationReceiverComponent>();
+        foreach (var (geiger, receiver) in query)
+        {
+            if (!geiger.IsEnabled)
+                continue;
```

Comment:
Server shouldn't be iterating over inactive components.

Reply:
I remove this redundant check. The combination of `GeigerComponent` and `RadiationReceiverComponent` should appear only on active geiger counters.